### PR TITLE
CIV-5797 - Transform Mediation model - CCD

### PR DESCRIPTION
**Pull request by portizdev**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-5797
https://tools.hmcts.net/jira/browse/CIV-7015

### Change description ###
Transform Mediation Model into CCD Model

BUGS fixed:

- Mediation phone number (can-we-use)
- Confirm your details
- Save preferred bilingual language

**Framework developed by @kenneth-hmcts .**

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/routes/features/dashboard/claimSummaryController.ts [hunk 1]
```diff
@@ -18,6 +19,7 @@ claimSummaryController.get([DEFENDANT_SUMMARY_URL], async (req, res, next: NextF
     if (claim && !claim.isEmpty()) {
       const latestUpdateContent = getLatestUpdateContent(claimId, claim);
       const documentsContent = getDocumentsContent(claim, claimId);
+      await saveDraftClaim(claim.id, claim);
```
Comment (portizdev) [inline src/main/routes/features/dashboard/claimSummaryController.ts:22]: Fix for saving in bilingual-preference. The claim was not being saved (on claim summary page (Respond to claim)).

File: src/main/routes/features/mediation/mediationIndividualPhoneController.ts [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-import {GenericYesNo} from '../../../common/form/models/genericYesNo';
+import {GenericYesNo} from 'common/form/models/genericYesNo';
 
 const mediationIndividualPhoneViewPath = 'features/mediation/can-we-use';
 const mediationIndividualPhoneController = Router();
 
 async function renderView(form: GenericForm<MediationIndividualPhoneNumber>, res: Response, claimId: string): Promise<void> {
   const claim: Claim = await getCaseDataFromStore(claimId);
-  res.render(mediationIndividualPhoneViewPath, {form, respondentTelNumber: claim.respondent1?.partyPhone});
+  res.render(mediationIndividualPhoneViewPath, {form, respondentTelNumber: claim.respondent1?.partyPhone?.phone});
```
Comment (portizdev) [inline src/main/routes/features/mediation/mediationIndividualPhoneController.ts:18]: Fix https://tools.hmcts.net/jira/browse/CIV-7015

File: src/main/views/features/mediation/can-we-use.njk [hunk 3]
```diff
@@ -16,7 +16,7 @@
     <main class="govuk-main-wrapper" role="main">
       {% set title = t('PAGES.MEDIATION_CAN_WE_USE.CONFIRM_YOUR_NUMBER') %}
       {% set titleEnterPhone = t('PAGES.MEDIATION_CAN_WE_USE.ENTER_PHONE_NUMBER') %}
-      {% set pText = t('PAGES.MEDIATION_CAN_WE_USE.CAN_MEDIATION_USE', {respondentTelNumber: respondentTelNumber.phone}) %}
+      {% set pText = t('PAGES.MEDIATION_CAN_WE_USE.CAN_MEDIATION_USE', {respondentTelNumber: respondentTelNumber}) %}
```
Comment (portizdev) [inline src/main/views/features/mediation/can-we-use.njk:19]: Fix https://tools.hmcts.net/jira/browse/CIV-7015

File: src/main/views/features/response/citizenDetails/citizen-details.njk [hunk 4]
```diff
@@ -12,17 +12,17 @@
 {% macro citizenDetailsSection(t, type) %}
   {% if type === partyType.INDIVIDUAL %}
     <h2 class="govuk-heading-m">{{ t('COMMON.INPUT_LABELS.TITLE') }}</h2>
-    <p class="govuk-body">{% if party.model.individualTitle %}{{ party.model.individualTitle }}{% else %} individualTitle Test{% endif %}</p>
+    <p class="govuk-body">{% if party.model.individualTitle %}{{ party.model.individualTitle }}{% endif %}</p>
```
Comment (portizdev) [inline src/main/views/features/response/citizenDetails/citizen-details.njk:15]: In confirm your details when there was no title (optional). Texts like "individualTitle test" were being displayed.

**Review by LeonardoPalmeiro — APPROVED** [review]
LGTM

File: src/main/routes/features/dashboard/claimSummaryController.ts (hunk 1 above)
Comment (TatsyMaslova) [reply src/main/routes/features/dashboard/claimSummaryController.ts:22]: Are you sure this is needed? we save the claim when the defendant clicks on the respond to claim button. 
In this case what if claim is already in draft store? we might then override whatever is already there and potentially a draft response. We have a utility function that Sampan created - which first tries to get the claim from store and then if it does not exist gets it from civil service and saves it into draft store.. However we save draft claim anyways when we click on respond to claim. and i think we agreed that  we always have to have the latest update from civil service here. So i am worried that this change will introduce a lot of bugs for Release1

**Review by hmcts-version1-ignacio — DISMISSED** [review]

File: src/main/routes/features/dashboard/claimSummaryController.ts (hunk 1 above)
Comment (portizdev) [reply src/main/routes/features/dashboard/claimSummaryController.ts]: Solve, please review it again <3

**Review by TatsyMaslova — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Review by kenneth-hmcts — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Pull request merged**